e2e/registration_test.go

```diff
@@ -140,7 +140,7 @@ func TestE2E_ServiceRegistration_DeregisterWhenStopped(t *testing.T) {
 	config := baseConfig(tempDir).appendID(id).
 		appendConsulBlock(srv).appendTerraformBlock().
 		appendModuleTask("disabled_task", "mkam/hello/cts",
-			"enabled = false") // optimization since task running is not relevant to test
+			"enabled = true")
```

Comment:
This test used to work with false because registration was performed after once mode. The test waits for the registration to complete before trying to stop CTS. With registration after once, and all the tasks disabled, the `cts.WaitForTestReadiness(defaultWaitForTestReadiness)` function is not working as expected.

For now I am enabling this task so that the `WaitForTestReadiness` works as intended. Will have to revisit why this doesn't work with all disabled tasks later

Reply:
I poked around because I was curious as to what's going on and here's my findings:

`WaitForTestReadiness` looks through all tasks and checks only enabled tasks. Since the only task is disabled, it will essentially return immediately. Because it's ignoring the disabled tasks, it doesn't wait for the task to finish initializing, which is preventing the graceful shutdown.

```
        2022-05-20T09:42:04.151-0500 [ERROR] driver.terraform: error initializing workspace for task: task_name=disabled_task
        2022-05-20T09:42:04.151-0500 [ERROR] ctrl: error initializing task: task_name=disabled_task
        2022-05-20T09:42:04.151-0500 [DEBUG] ctrl: cleaned up task that errored initializing: task_name=disabled_task
        2022-05-20T09:42:04.151-0500 [ERROR] cli: error running controller: error="error tf-init for 'disabled_task': context canceled"
```
This wasn't an issue when we registered tasks after once mode because of `WaitForConsulServiceRegistered`. This would wait for CTS to be registered, which would only happen after the task was fully initialized.

Reply:
So having it set to true is the way to go since I don't think we can determine whether a task is initialized or not in the API. It was a very minor optimization anyway!